packages/flame/test/components/ellipse_component_test.dart

```diff
@@ -0,0 +1,62 @@
+import 'package:flame/components.dart';
+import 'package:flame/geometry.dart';
+import 'package:flame/src/geometry/ellipse_component.dart';
+import 'package:flutter_test/flutter_test.dart';
+
+void main() {
+  group('EllipseComponent', () {
+    test('width and height setters and getters', () {
+      final component = EllipseComponent(width: 100, height: 50);
+      expect(component.width, 100);
+      expect(component.height, 50);
+
+      component.width = 200;
+      component.height = 100;
+      expect(component.width, 200);
+      expect(component.height, 100);
+    });
+
+    test('relative sizing', () {
+      final parentSize = Vector2(300, 150);
+      final component = EllipseComponent.relative(
+        0.5,
+        0.4,
+        parentSize: parentSize,
+      );
+
+      expect(component.size.x, 150);
+      expect(component.size.y, 60);
+    });
+
+    test('containsPoint', () {
+      final component = EllipseComponent(width: 100, height: 50)
+        ..position = Vector2(150, 150) // 中心放置在(150, 150)
+        ..anchor = Anchor.center; // 确保以中心为锚点
+
+      expect(component.containsPoint(Vector2(150, 150)), isTrue); // 在中心
+      expect(component.containsPoint(Vector2(100, 100)), isFalse); // 不在椭圆内
+      expect(component.containsPoint(Vector2(200, 150)), isTrue); // 在边界上
+    });
+
+    test('containsLocalPoint', () {
+      final component = EllipseComponent(width: 100, height: 50);
+      expect(component.containsLocalPoint(Vector2(50, 25)), isTrue);
+      expect(component.containsLocalPoint(Vector2(100, 50)), isFalse);
+    });
+
+    test('lineSegmentIntersections', () {
+      final component = EllipseComponent(width: 100, height: 50)
+        ..position = Vector2(100, 50)
+        ..anchor = Anchor.center;
+
+      final lineSegment = LineSegment(
+        Vector2(50, 50),
+        Vector2(150, 50),
+      );
+
+      final intersections = component.lineSegmentIntersections(lineSegment);
+      expect(intersections.length, 2);
+      expect(intersections[0].x < intersections[1].x, isTrue);
```

Comment:
Can you check the approximate intersection points here too? 
You can use the `closeToVector` matcher, like here: https://github.com/flame-engine/flame/blob/267d6801cb7e6cbbaa450e24e38aaa7d8fcfc03f/packages/flame/test/collisions/collision_detection_test.dart#L1037

Reply:
It's very difficult to write a class similar to CircleCircleIntersections for ellipses. I tried but couldn't figure it out. It seems like the EllipseComponent I wrote isn't very useful.
the CircleCircleIntersections code as bellow
···dart
class CircleCircleIntersections
    extends Intersections<CircleComponent, CircleComponent> {
  @override
  Set<Vector2> intersect(CircleComponent shapeA, CircleComponent shapeB) {
    final centerA = shapeA.absoluteCenter;
    final centerB = shapeB.absoluteCenter;
    final distance = centerA.distanceTo(centerB);
    final radiusA = shapeA.radius;
    final radiusB = shapeB.radius;
    if (distance > radiusA + radiusB) {
      // Since the circles are too far away from each other to intersect we
      // return the empty set.
      return {};
    } else if (distance < (radiusA - radiusB).abs()) {
      // When one circle is contained within the other there is only a collision
      // if the outer circle isn't hollow.
      final outerShape = radiusA > radiusB ? shapeA : shapeB;
      if (outerShape.isSolid) {
        final center = outerShape == shapeA ? centerB : centerA;
        return {center};
      } else {
        return {};
      }
    } else if (distance == 0 && radiusA == radiusB) {
      // The circles are identical and on top of each other, so there are an
      // infinite number of solutions. Since it is problematic to return a
      // set of infinite size, we'll return 4 distinct points here.
      return {
        shapeA.absoluteCenter + Vector2(radiusA, 0),
        shapeA.absoluteCenter + Vector2(0, -radiusA),
        shapeA.absoluteCenter + Vector2(-radiusA, 0),
        shapeA.absoluteCenter + Vector2(0, radiusA),
      };
    } else {
      // There are definitely collision points if we end up in here.
      // To calculate these we use the fact that we can form two triangles going
      // between the center of shapeA, the point in between the shapes which the
      // intersecting line goes through, and then two different triangles are
      // formed with the two intersection points as the last corners.
      // The length to the point in between the circles is first calculated,
      // this is [lengthA], then we calculate the length of the other cathetus
      // [lengthB]. Then the [centerPoint] is calculated, which is the point
      // which the intersecting line goes through in between the shapes.
      // At this point we know the two first points of the triangles, the center
      // of [shapeA] and the [centerPoint], the two third points of the
      // different triangles are the intersection points that we are looking for
      // and we get those points by calculating the [delta] from the
      // [centerPoint] to the intersection points.
      // The result is then [centerPoint] +- [delta].
      final lengthA = (pow(radiusA, 2) - pow(radiusB, 2) + pow(distance, 2)) /
          (2 * distance);
      final lengthB = sqrt((pow(radiusA, 2) - pow(lengthA, 2)).abs());
      final centerPoint = shapeA.absoluteCenter +
          (shapeB.absoluteCenter - shapeA.absoluteCenter) * lengthA / distance;
      final delta = Vector2(
        lengthB *
            (shapeB.absoluteCenter.y - shapeA.absoluteCenter.y).abs() /
            distance,
        -lengthB *
            (shapeB.absoluteCenter.x - shapeA.absoluteCenter.x).abs() /
            distance,
      );
      return {
        centerPoint + delta,
        centerPoint - delta,
      };
    }
  }
}
···

Reply:
Yeah, it's a quite hard problem I guess.
Should we close this PR then, or do you want to keep trying?

Reply:
yeah, you can close it sry about that

Reply:
No problem, next person that wants to try can build on your PR :)